client/src/components/location.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+class Location extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    }
+    this.slideRight = this.slideRight.bind(this);
+    this.slideLeft = this.slideLeft.bind(this);
+  }
+
+  slideRight(event) {
+    event.preventDefault();
+    this.setState(prevState => {
```

Comment:
I haven't seen this syntax before for grabbing the previous state, learn something new everyday!